components/script/dom/bluetooth.rs

```diff
@@ -686,20 +686,24 @@ impl PermissionAlgorithm for Bluetooth {
     #[allow(unrooted_must_root)]
     // https://webbluetoothcg.github.io/web-bluetooth/#revoke-bluetooth-access
     fn permission_revoke(_descriptor: &BluetoothPermissionDescriptor, status: &BluetoothPermissionResult) {
-        // Step 1. TODO
+        // Step 1.
+        let allowed_devices = get_allowed_devices();
+        // Step 2.
         let bluetooth = status.get_bluetooth();
         let device_map = bluetooth.get_device_map().borrow();
-        // Step 2.
         for (id, device) in device_map.iter() {
             let id = DOMString::from(id.clone());
             // Step 2.1.
-            if let Some(_allowed_device) = find_allowed_device(id) {
-                if let Some(_represented_device) = device.get_represented_device() {
-                    //TODO Set allowed_services
-                    continue;
-                }
+            if allowed_devices.iter().find(|d| d.deviceId == id).is_some() &&
```

Comment:
Using `any` instead of `find` and `is_some`  can make this a bit shorter.
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.any